store/displayPreferences.ts

```diff
@@ -134,170 +98,119 @@ export const actions: ActionTree<
+      await dispatch(
+        'snackbar/pushSnackbarMessage',
+        { message, color: 'error' },
+        { root: true }
+      );
     }
   },
-
-  /**
-   * On query success, stores the result and call the callbacks
-   *
-   * @param {any} param0 - Vuex
-   * @param {any} param0.commit - Vuex commit
-   * @param {any} param0.dispatch - Vuex dispatch
-   * @param {any} param1 - Payload
-   * @param {DisplayPreferencesDto} param1.displayPreferences - Display preferences object
-   */
-  async initStateSuccess(
-    { commit, dispatch },
-    { displayPreferences }: { displayPreferences: DisplayPreferencesDto }
-  ) {
-    commit('INIT_STATE', { displayPreferences });
-    await dispatch('callAllCallbacks');
-  },
-
-  /**
-   * On query error, sends the error and message to the store logger
-   *
-   * @param {any} param0 - Vuex
-   * @param {any} param0.dispatch - Vuex dispatch
-   * @param {any} param1 - Payload
-   * @param {any} param1.error - Try-catch error
-   */
-  async initStateFailure({ dispatch }, { error }: { error: unknown }) {
-    const message = this.$i18n.t('failedRetrievingDisplayPreferences');
-    await dispatch('requestError', { message, error });
-  },
-
   /**
    * Pushes the current state to the server
    *
-   * @param {any} param0 - Vuex
-   * @param {any} param0.state - Vuex state
-   * @param {any} param0.dispatch - Vuex dispatch
+   * @param {any} context - Vuex action context
+   * @param {any} context.state - Vuex state
+   * @param {any} context.dispatch - Vuex dispatch
    */
-  async pushState({ state, dispatch }) {
+  async pushState({ state, dispatch, commit }) {
+    commit('SYNCING_STARTED');
     try {
+      // The fetch part is done because DisplayPreferences doesn't accept partial updates
+      const responseFetch = await this.$api.displayPreferences.getDisplayPreferences(
+        {
+          displayPreferencesId: 'usersettings',
+          userId: this.$auth.user?.Id,
+          client: 'vue'
+        }
+      );
+
+      if (responseFetch.status !== 200)
+        throw new Error(
+          'get display preferences status response = ' + responseFetch.status
+        );
+
+      const displayPrefs = responseFetch.data;
+      displayPrefs.CustomPrefs = {};
+      Object.assign(displayPrefs.CustomPrefs, state);
+      for (const [key, value] of Object.entries(displayPrefs.CustomPrefs)) {
+        /**
+         * As TypeScript typings are erased at runtime, we can't rely on typings to distinguish between
+         * the variables that belong just to the store and the ones that we want to be synced to the server.
+         *
+         * We can overcome this limitation in the future by storing our prefs in another store and leaving this store
+         * just for the API calls and the state of syncing.
+         */
+        if (key === 'syncing') {
+          delete displayPrefs.CustomPrefs[key];
+          continue;
+        }
```

Comment:
I'd just delete the key before the for :)

Reply:
How though?

Reply:
`delete displayPrefs.CustomPrefs?.syncing;`?

Reply:
Honestly, I don't know what I was thinking by asking that.